lib/pdf/reader/xref.rb

```diff
@@ -123,6 +123,13 @@ def load_offsets(offset = nil)
     def load_xref_table(buf)
       params = []
 
+      # HACK: some PDF writers start numbering a 1 instead of 0. Fix up the number.
+      tokens = buf.instance_variable_get(:@tokens)
+      if tokens[0] == '1' and tokens[2, 3] == ['0000000000', '65535', 'f']
+        # TODO should this be logged?
+        tokens[0] = "0"
+      end
```

Comment:
What about something like this?

```suggestion
      params = [buf.token, buf.token]
                                                                       
      # Some PDF writers start numbering a 1 instead of 0. Fix up the number.
      if params[0] == "1"
        params[0] = "0"
      end
```

Reply:
I don't think it's safe to unconditionally change object 1 to object 0.

There can be multiple sections in the xref table; AFAIK contiguous sections don't have to be merged, so it would be possible to have the following layout:

xref
0 1
...
1 n
...
...

Latest commit should address your concern